stl/src/locale0.cpp

```diff
@@ -136,7 +138,12 @@ __PURE_APPDOMAIN_GLOBAL locale::id ctype<unsigned short>::id(0);
 __PURE_APPDOMAIN_GLOBAL locale::id codecvt<unsigned short, char, mbstate_t>::id(0);
 
 _MRTIMP2_PURE const locale& __CLRCALL_PURE_OR_CDECL locale::classic() { // get reference to "C" locale
-    _Init();
+    const auto ptr = reinterpret_cast<locale::_Locimp*>(InterlockedCompareExchangePointerAcquire(

```

Comment:
@strega-nil-ms had an excellent observation - we need to check whether (1) `locale0.cpp` is compiled for `/clr:pure` in the internal build system (I believe it is), and (2) whether `InterlockedCompareExchangePointerAcquire()` is one of the rare intrinsics that is actually supported for `/clr:pure` (I suspect that it is not). If this machinery is incompatible with `/clr:pure`, we'll need `#ifdef _M_CEE_PURE` to handle that.

Reply:
I think that this ibtrinsic is worst choice anyway, see my above comments